cmake/DefaultTargetProperties.cmake

```diff
@@ -83,15 +82,11 @@ function(set_default_target_properties target_name)
     #-------------------------------------------------------------------
     # RPATH wrangling.
 
-    # With the 'usual' install path structure in mind, search from the
-    # bin directory (i.e. a binary loading a dynamic library) and then
-    # from the current directory (i.e. dynamic library loading another
-    # dynamic library).
     if (APPLE)
         # TODO(DF): handle RPATH on OSX - e.g. @rpath vs. @loader_path
         message(AUTHOR_WARNING "OSX RPATH not configured")
     else ()
```

Comment:
[question] Does the project supports Windows?
Perhaps it wouldn't be too painful to set `$ORIGIN` behind an `elseif(UNIX)`

Reply:
Yeah we will definitely be supporting Windows at some point.  Full Windows support is for another ticket, but we've been trying to ensure a reasonable foundation.  I updated the OS-specific branching in 74580c30 (as well as reworded some comments).